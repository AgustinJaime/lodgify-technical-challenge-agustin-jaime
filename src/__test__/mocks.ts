const fetchedDataMock = [
  {
    name: "Group 1",
    tasks: [
      {
        description: "Task 1 - 1",
        value: 10,
        checked: true,
      },
      {
        description: "Task 2 - 1",
        value: 20,
        checked: false,
      },
    ],
  },
  {
    name: "Group 2",
    tasks: [
      {
        description: "Task 1 - 1",
        value: 30,
        checked: false,
      },
      {
        description: "Task 2 - 1",
        value: 20,
        checked: false,
      },
    ],
  },
  {
    name: "Group 3",
    tasks: [
      {
        description: "Task 1 - 1",
        value: 20,
        checked: false,
      },
    ],
  },
];

export const mockInitialState = {
  tasks: {
    loading: "SUCCEEDED",
    data: fetchedDataMock,
  },
};
