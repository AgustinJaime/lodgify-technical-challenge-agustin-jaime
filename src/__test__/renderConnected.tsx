import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import reducers from "../store/reducers";

const renderConnected = (
  ui: React.ReactElement,
  initialState = {},
  {
    store = configureStore({
      reducer: reducers,
      preloadedState: initialState,
    }),
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }: React.PropsWithChildren) => (
    <Provider store={store}>{children}</Provider>
  );

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export default renderConnected;
