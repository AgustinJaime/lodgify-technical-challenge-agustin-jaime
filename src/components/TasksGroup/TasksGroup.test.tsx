import { mockInitialState } from "../../__test__/mocks";
import renderConnected from "../../__test__/renderConnected";
import TasksGroups from "./TasksGroup";

const props = {
  group: mockInitialState.tasks.data[0],
  groupIndex: 1,
  expanded: true,
  handleCheckBoxClick: jest.fn(),
  handleExpandGroup: jest.fn(),
  completed: false,
};

describe("<App>", () => {
  it("should render without issues", () => {
    const documentBody = renderConnected(<TasksGroups {...props} />);
    expect(documentBody).toMatchSnapshot();
  });
  it("should render group name", () => {
    const documentBody = renderConnected(<TasksGroups {...props} />);
    documentBody.getByText("Group 1");
  });
  it("should render 'show'", () => {
    const documentBody = renderConnected(<TasksGroups {...props} />);
    documentBody.getByText("Show");
  });
  it("should render tasks", () => {
    const documentBody = renderConnected(<TasksGroups {...props} />);
    const tasks = documentBody.queryAllByTestId("taskTestId_", {
      exact: false,
    });
    expect(tasks).toHaveLength(2);
  });
  it("should render input with description and checked as default", () => {
    const documentBody = renderConnected(<TasksGroups {...props} />);
    const taskOne = documentBody.queryByRole("checkbox", {
      name: "Task 1 - 1",
    });
    const taskTwo = documentBody.queryByRole("checkbox", {
      name: "Task 2 - 1",
    });
    expect(taskOne).toBeInTheDocument();
    expect(taskTwo).toBeInTheDocument();
    expect(taskOne).toBeChecked();
    expect(taskTwo).not.toBeChecked();
  });
});
