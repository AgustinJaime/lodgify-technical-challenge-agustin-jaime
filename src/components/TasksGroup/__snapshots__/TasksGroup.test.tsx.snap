// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should render without issues 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="group-ctn"
      >
        <div
          class="group-header"
          data-testid="groupHeaderTestId_1"
        >
          <div
            class="group-header-section"
          >
            <svg
              class="list-i"
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
              />
            </svg>
            <p
              class=""
            >
              Group 1
            </p>
          </div>
          <div
            class="group-header-section"
          >
            <p>
              Show
            </p>
            <svg
              class="arrow-i expanded"
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
              />
            </svg>
          </div>
        </div>
        <div
          class="group-list"
          data-testid="taskTestId_0"
        >
          <label>
            <input
              checked=""
              class="list-i"
              type="checkbox"
            />
            Task 1 - 1
          </label>
        </div>
        <div
          class="group-list"
          data-testid="taskTestId_1"
        >
          <label>
            <input
              class="list-i"
              type="checkbox"
            />
            Task 2 - 1
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="group-ctn"
    >
      <div
        class="group-header"
        data-testid="groupHeaderTestId_1"
      >
        <div
          class="group-header-section"
        >
          <svg
            class="list-i"
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
            />
          </svg>
          <p
            class=""
          >
            Group 1
          </p>
        </div>
        <div
          class="group-header-section"
        >
          <p>
            Show
          </p>
          <svg
            class="arrow-i expanded"
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="group-list"
        data-testid="taskTestId_0"
      >
        <label>
          <input
            checked=""
            class="list-i"
            type="checkbox"
          />
          Task 1 - 1
        </label>
      </div>
      <div
        class="group-list"
        data-testid="taskTestId_1"
      >
        <label>
          <input
            class="list-i"
            type="checkbox"
          />
          Task 2 - 1
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
